recipes/uhd/meta.yaml

```diff
@@ -0,0 +1,98 @@
+{% set version = "3.14.1.1" %}
+
+package:
+  name: uhd
+  version: {{ version }}
+
+source:
+  url: https://github.com/EttusResearch/uhd/archive/v{{ version }}.tar.gz
+  sha256: 8cbcb22d12374ceb2859689b1d68d9a5fa6bd5bd82407f66952863d5547d27d0
+  patches:
+    - 0001-Fix-snprintf-redeclared-as_snprintf.patch
+    - 0001-lib-utils-Don-t-use-hard-coded-path-constants.patch
+    - 0001-python-change-CMake-variable-for-library-ext.patch
+    - 0002-python-Fix-internal-library-name-incl.-suffix-to-mat.patch
+    - 0003-python-Do-not-link-against-python-lib-for-building-a.patch
+    - 0004-Fix-build-RPATH-remaining-on-installed-libpyuhd-by-n.patch
+
+build:
+  number: 0
+  run_exports:
+    - {{ pin_subpackage('uhd', min_pin='x.x.x', max_pin='x.x.x') }}
+  # windows requires vs2015+ to build and therefore can't build against py2k
+  skip: True  # [win and py2k]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake >=3.5.1
+    - jom  # [win]
+    - {{ cdt('libudev-devel') }}  # [linux]
+    - pkg-config  # [not win]
+
+  host:
+    - boost
+    - boost-cpp
+    - libusb >=1
+    - mako >=0.5.0
+    - ncurses  # [not win]
+    - numpy
+    - python
+    - requests
+    - setuptools
+
+  run:
+    - boost
+    - boost-cpp
```

Comment:
I think this only contains headers so you won’t need it at runtime?

Reply:
It looks like boost-cpp is a dep of boost so it gets pulled in no matter what.

https://github.com/conda-forge/boost-feedstock/blob/master/recipe/meta.yaml#L28

Only actual deps of `uhd` should be listed. Any deps of deps can be excluded.

Reply:
They are both definitely needed in host for the build (although the Boost Python requirement should be going away with the next release), and the info after the build spits out a lot of "needed DSO" pertaining to ``boost-cpp`` and one pertaining to ``boost``. So I think both are directly required as a run dependency of ``uhd``. Please advise if not though.

Reply:
Sure sounds good to me! As long as we are not pinning the versions and letting the global pins take care of it, there is no harm in keeping them. The conda build warnings are of mixed use in my experience. Many codes overlink and so I tend to ignore things but maybe I shouldn't!

Reply:
Ah, good for me to keep in mind. But yes, in this case I think it's correct.